File: studyhub-api/src/main/java/com/studyhub/studyhub_api/service/statistics/StatisticsService.java
package com.studyhub.studyhub_api.service.statistics;

import com.studyhub.studyhub_api.dto.response.statistics.RevenueStatisticsResponse;
import com.studyhub.studyhub_api.dto.response.statistics.StatisticsBasicResponse;

import java.util.List;

public interface StatisticsService {
    StatisticsBasicResponse getBasicStatistics();
    List<RevenueStatisticsResponse> getRevenueStatistics(Integer year);
}

File: StudyHub_ag/src/app/_service/statistics/statistics.service.ts
import { Injectable } from '@angular/core';
import { BaseService } from '../base/base.service';
import { HttpClient } from '@angular/common/http';
import { API_ENDPOINTS } from '../base/api-endpoints';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root',
})
export class StatisticsService extends BaseService<CourseLiteProjection> {
  constructor(http: HttpClient) {
    super(http, API_ENDPOINTS.STATISTICS);
  }

  getBasicStatistics(): Observable<ApiResponse<StatisticsBasicResponse>> {
    return this.customRequest('GET', API_ENDPOINTS.STATISTICS_ENDPOINTS.BASIC);
  }

  getRevenueStatisticsByYear(
    year = 2026,
  ): Observable<ApiResponse<RevenueStatisticsResponse[]>> {
    return this.customRequest(
      'GET',
      `${API_ENDPOINTS.STATISTICS_ENDPOINTS.REVENUE_YEAR}/${year}`,
    );
  }
}
